nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -131,7 +131,8 @@ def write_entry(profile: str | None, generation: int, specialisation: str | None
         specialisation=" (%s)" % specialisation if specialisation else "")
 
     try:
-        subprocess.check_call([bootspec.initrdSecrets, "@efiSysMountPoint@%s" % (initrd)])
+        if bootspec.initrdSecrets is not None:
+            subprocess.check_call([bootspec.initrdSecrets, "@efiSysMountPoint@%s" % (initrd)])
     except FileNotFoundError:
         pass
```

Comment:
Since this shouldn't happen unless there is an actual initrd secrets thing to call, we should remove this branch so we don't squash actual issues with the system config (if they have initrd secrets, the initrd secrets tool should exist).

Reply:
Right, good point

Reply:
Done.